src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -180,37 +215,44 @@ private async Task<long> ReplayTaggedMessagesAsync(ReplayTaggedMessages replay)
 
             // Need to know what the highest seqNo of this query will be
             // and return that as part of the RecoverySuccess message
-            var maxSeqNoEntry = await _journalCollection.Value.Find(seqNoFilter)
-                .SortByDescending(x => x.Ordering)
-                .Limit(1)
-                .SingleOrDefaultAsync();
-
-            if (maxSeqNoEntry == null)
-                return 0L; // recovered nothing
 
-            var maxOrderingId = maxSeqNoEntry.Ordering.Value;
-            var toSeqNo = Math.Min(toSequenceNr, maxOrderingId);
-
-            var readFilter = builder.AnyEq(x => x.Tags, tag);
-            if (fromSequenceNr > 0)
-                readFilter &= builder.Gt(x => x.Ordering, new BsonTimestamp(fromSequenceNr));
-            if (toSequenceNr != long.MaxValue)
-                readFilter &= builder.Lte(x => x.Ordering, new BsonTimestamp(toSeqNo));
-            var sort = Builders<JournalEntry>.Sort.Ascending(x => x.Ordering);
-
-            await _journalCollection.Value
-                .Find(readFilter)
-                .Sort(sort)
-                .Limit(limitValue)
-                .ForEachAsync(entry =>
-                {
-                    var persistent = ToPersistenceRepresentation(entry, ActorRefs.NoSender);
-                    foreach (var adapted in AdaptFromJournal(persistent))
-                        replay.ReplyTo.Tell(new ReplayedTaggedMessage(adapted, tag, entry.Ordering.Value),
-                            ActorRefs.NoSender);
-                });
+            var (perCallCts, unitedCts) = CreatePerCalLCts();
+            using (perCallCts)
+            using (unitedCts)
```

Comment:
Use same CTS for both database calls, since this is part of a single "scope" from the PoV of the caller.